sdv/tabular/base.py

```diff
@@ -185,6 +187,62 @@ def sample(self, num_rows=None, max_retries=100):
 
         return sampled.head(num_rows)
 
+    def _make_conditions_df(self, conditions, num_rows):
+        if isinstance(conditions, pd.Series):
+            conditions = pd.DataFrame([conditions] * num_rows)
+
+        elif isinstance(conditions, dict):
+            try:
+                conditions = pd.DataFrame(conditions)
+            except ValueError:
+                conditions = pd.DataFrame([conditions] * num_rows)
+
+        elif not isinstance(conditions, pd.DataFrame):
+            raise TypeError("`conditions` must be a dataframe, a dictionary or a pandas series.")
+
+        return conditions
+
+    def sample(self, num_rows=None, max_retries=100, conditions=None):
+        """Write this."""
+        if conditions is None:
+            return self._sample_conditioned_rows(num_rows, max_retries, None)
+
+        # convert conditions to dataframe
+        conditions = self._make_conditions_df(conditions, num_rows)
+
+        # transform conditions, will use RDT when that's ready
```

Comment:
RDT should be ready for it, but I suspect that we will need to also tackle the `Table.transform` and `Table.reverse_transform` methods to fully support this.
This comment can also be removed, since the loop will stay as it is.